src/reducers/utils.js

```diff
@@ -0,0 +1,50 @@
+const checkItemExist = (list, productId) => {
+  const res = list.filter(
+    ({ id, status }) =>
+      id === productId && (status ? status !== "deleted" : true)
+  );
+  return res.length > 0 ? true : false;
+};
+
+const appendItem = (list, product, wish) => {
+  return checkItemExist(list, product.id)
+    ? wish === "wish"
```

Comment:
Avoid using nested ternary operator. I think Tanay has a tweet on this. Check this: https://twitter.com/tanaypratap/status/1377334331853574145
I can't put it in better words :)